### PR TITLE
Remove Teleport 9 Deprecations

### PR DESCRIPTION
**Pull request by zmb3**
Removes a few items marked for deprecation in Teleport 9.

For host certs, it looks like the client side did not properly handle both old/new formats, so we need to push that out and delete in Teleport 10 instead.

### CODE REVIEW

**Review by Joerger — APPROVED** [review]

**Comment by russjones** [comment]
@zmb3 Yes, we're dropping support for CentOS 6 in Teleport 9.

File: lib/web/apiserver.go [hunk 1]
```diff
@@ -2337,7 +2337,7 @@ func (h *Handler) hostCredentials(w http.ResponseWriter, r *http.Request, p http
 		return nil, trace.Wrap(err)
 	}
 
-	return auth.LegacyCertsFromProto(certs), nil
+	return certs, nil
```
Comment (Joerger) [inline lib/web/apiserver.go:2340]:
```suggestion
	// Teleport 8 clients are still expecting the legacy JSON format.
	// Teleport 9 clients handle both legacy and new.
	// TODO(zmb3) return certs directly in Teleport 10
	return auth.LegacyCertsFromProto(certs), nil
```

**Review by Joerger — APPROVED** [review]

File: lib/web/apiserver.go (hunk 1 above)
Comment (zmb3) [reply lib/web/apiserver.go:2340]: I don't understand what you're suggesting.

**Review by greedy52 — APPROVED** [review]

File: lib/auth/apiserver.go [hunk 2]
```diff
@@ -150,9 +150,7 @@ func NewAPIServer(config *APIConfig) (http.Handler, error) {
 	srv.DELETE("/:version/tunnelconnections/:cluster", srv.withAuth(srv.deleteTunnelConnections))
 	srv.DELETE("/:version/tunnelconnections", srv.withAuth(srv.deleteAllTunnelConnections))
 
-	// Server Credentials
-	srv.POST("/:version/server/credentials", srv.withAuth(srv.generateHostCerts))
```
Comment (r0mant) [inline lib/auth/apiserver.go:154]: Can v8 clients call this method? If we're removing it from client side in v9, shouldn't we keep the server API until v10?

Comment (zmb3) [reply lib/auth/apiserver.go:154]: v8 clients will try GRPC first, and only attempt to call this if GRPC is not implemented. v8 servers have the GRPC, so the only way a v8 client could call this if if they hit a v7 server (which we don't allow, as you have to upgrade proxy/auth before nodes).

**Review by r0mant — APPROVED** [review]

**Pull request merged**